### PR TITLE
feat(Text): Migrate to variance and add new semantic variants [GM-204]

### PR DESCRIPTION
**Pull request by codecaaron**
### Overview

Variance Migration Doc: https://www.notion.so/codecademy/Variance-Migration-a675bbc31b584fec9dfff72c0364b194

* Adds updated Typography Docs to the Foundations page.
* Restructure + Rewrites the Text component documentation patterns for problem solving. 
* Adds second level of visual variants that match up loosely to html tags, but are specific to our own core Typography Scales.  `title-1` `p-base` etc.
* Sets progressive override structure for `<Text />`
  1. Tag styles
  2. Scale Styles
  3. Prop Styles
* Converts all props to `variance` props and updates their usages.

### PR Checklist

- [x] Related to designs: https://www.figma.com/file/rNsdbOwlw6L0ea3uAJrrz7
- [x] Related to JIRA ticket: GM-204
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.39-alpha.c7a752.0<br/>@codecademy/gamut-labs@14.0.1-alpha.c7a752.0<br/>@codecademy/gamut@32.0.1-alpha.c7a752.0<br/>@codecademy/styleguide@36.0.1-alpha.c7a752.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://607f5d998616ef00d52a9280--gamut-preview.netlify.app>https://607f5d998616ef00d52a9280--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/607f5d998616ef00d52a9280>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.39-alpha.df0407.0<br/>@codecademy/gamut-labs@14.0.1-alpha.df0407.0<br/>@codecademy/gamut@32.0.1-alpha.df0407.0<br/>@codecademy/styleguide@36.0.1-alpha.df0407.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://607f5f8439dd56039790fee8--gamut-preview.netlify.app>https://607f5f8439dd56039790fee8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/607f5f8439dd56039790fee8>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.39-alpha.181d24.0<br/>@codecademy/gamut-labs@14.0.1-alpha.181d24.0<br/>@codecademy/gamut@32.0.1-alpha.181d24.0<br/>@codecademy/styleguide@36.0.1-alpha.181d24.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://607f64d111f9a7009ebf1488--gamut-preview.netlify.app>https://607f64d111f9a7009ebf1488--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/607f64d111f9a7009ebf1488>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.39-alpha.20423d.0<br/>@codecademy/gamut-labs@14.0.1-alpha.20423d.0<br/>@codecademy/gamut@32.0.1-alpha.20423d.0<br/>@codecademy/styleguide@36.0.1-alpha.20423d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://607f7df7a274872e8fc626ce--gamut-preview.netlify.app>https://607f7df7a274872e8fc626ce--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/607f7df7a274872e8fc626ce>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.39-alpha.6e1384.0<br/>@codecademy/gamut-labs@14.0.1-alpha.6e1384.0<br/>@codecademy/gamut@32.0.1-alpha.6e1384.0<br/>@codecademy/styleguide@36.0.1-alpha.6e1384.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://607f7ff8f522f02d62e583d5--gamut-preview.netlify.app>https://607f7ff8f522f02d62e583d5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/607f7ff8f522f02d62e583d5>Deploy Logs</a></p>

File: packages/styleguide/stories/Foundations/Typography.stories.mdx [hunk 1]
```diff
@@ -16,6 +16,10 @@ import { LinkTo } from '~styleguide/blocks';
 
 For more checkout out our theme documentation for how to access and use all <LinkTo id="foundations-theme--font-family">typography design tokens</LinkTo>
 
+Our primary typeface is Aperçu Pro. It is a geometric-ish, humanist-ish sans-serif. Which basically means it gives us a voice that says “We are ruled by logic, but are creative and a bit unexpected as well”. It’s weight makes it attention-grabbing in dense layouts, and it’s legibility allows us to use it for both long-form articles and fine print.
```
Comment (dreamwasp) [inline packages/styleguide/stories/Foundations/Typography.stories.mdx:19]:
```suggestion
Our primary typeface is Aperçu Pro. It is a geometric-ish, humanist-ish sans-serif. Which basically means it gives us a voice that says “We are ruled by logic, but are creative and a bit unexpected as well”. Its weight makes it attention-grabbing in dense layouts, and its legibility allows us to use it for both long-form articles and fine print.
```

it's -> contraction, its -> possessive

**Review by dreamwasp — APPROVED** [review]
tiny grammar thing but this looks 🔥

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.39-alpha.088988.0<br/>@codecademy/gamut-labs@14.0.1-alpha.088988.0<br/>@codecademy/gamut@32.0.1-alpha.088988.0<br/>@codecademy/styleguide@36.0.1-alpha.088988.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608044bcdcf15600d4b8367a--gamut-preview.netlify.app>https://608044bcdcf15600d4b8367a--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608044bcdcf15600d4b8367a>Deploy Logs</a></p>

File: packages/gamut-labs/src/GlobalFooter/__tests__/GlobalFooter-test.tsx [hunk 2]
```diff
@@ -9,6 +9,8 @@ const renderView = setupRtl(GlobalFooter, {
 
 describe('GlobalFooter', () => {
   it('does not explode', () => {
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/GlobalFooter/__tests__/GlobalFooter-test.tsx:11]: This is no longer just a "does not explode" test 😄

File: packages/gamut-labs/src/landingPage/PageVideoGallery.tsx [hunk 3]
```diff
@@ -12,9 +12,10 @@ const Grid = styled.div`
   gap: 8px;
 
   ${mediaQueries.sm} {
-    // there must be a mathematical explanation for why ~12:5 is the correct ratio for
+    /** there must be a mathematical explanation for why ~12:5 is the correct ratio for
     // the column widths but I do not know what it is. Regardless, the effect is that
     // all three videos will maintain a 16:9 aspect ratio with an even gap between them.
+    */
```
Comment (JoshuaKGoldberg) [inline packages/gamut-labs/src/landingPage/PageVideoGallery.tsx:18]: Nit 😄 , unrelated?

File: packages/gamut/src/Typography/variants.ts [hunk 4]
```diff
@@ -0,0 +1,75 @@
+export const typographyStyleVariants = {
+  'title-1': {
```
Comment (JoshuaKGoldberg) [inline packages/gamut/src/Typography/variants.ts:2]: Can we use `xs`, `sm`, etc. on these? I don't like using the same terminology as the heading role sizes; people tend to mess up accessibility when we do.

File: packages/styleguide/stories/Foundations/Typography.stories.mdx [hunk 5]
```diff
@@ -16,6 +16,10 @@ import { LinkTo } from '~styleguide/blocks';
 
 For more checkout out our theme documentation for how to access and use all <LinkTo id="foundations-theme--font-family">typography design tokens</LinkTo>
 
+Our primary typeface is Aperçu Pro. It is a geometric-ish, humanist-ish sans-serif. Which basically means it gives us a voice that says “We are ruled by logic, but are creative and a bit unexpected as well”. Its weight makes it attention-grabbing in dense layouts, and its legibility allows us to use it for both long-form articles and fine print.
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Foundations/Typography.stories.mdx:19]:
```suggestion
Our primary typeface is Aperçu Pro. It is a geometric-ish, humanist-ish sans-serif. Its voice says _“we are ruled by logic, but are creative and a bit unexpected as well”_. Its weight makes it attention-grabbing in dense layouts, and its legibility allows us to use it for both long-form articles and fine print.
```

File: packages/styleguide/stories/Typography/Text.stories.mdx [hunk 6]
```diff
@@ -10,61 +14,112 @@ import { PropsTable } from '~styleguide/blocks';
...
+
+### Usage
+
+```tsx
+import { Text } from '@codecademy/gamut';
+
+<Text as="h4" variant="title-1" />;
+```
+
+**Note**: Be aware that you must replicate any default behavior if you override these props with responsive values.
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Typography/Text.stories.mdx:92]: Err, what does this mean?

File: packages/styleguide/stories/Typography/Text.stories.mdx [hunk 7]
```diff
@@ -10,61 +14,112 @@ import { PropsTable } from '~styleguide/blocks';
...
-          </Fragment>
-        ))}
-      </GridBox>
-    )}
+  <Story name="Text" args={{ children: 'Hello World' }}>
+    {(args) => <Text {...args} />}
   </Story>
 </Canvas>
+
+<PropsTable story="Text" defaultGroups={['typography']} />
```
Comment (JoshuaKGoldberg) [inline packages/styleguide/stories/Typography/Text.stories.mdx:125]: Tangentially related, is there a ticket somewhere for maybe adding the alias more prominently? 
Screenshot of the props table with unclear aliases

**Review by JoshuaKGoldberg — CHANGES_REQUESTED** [review]
The new docs are great, very nice!

Requesting changes on `-1` -> `-xs`?

File: packages/styleguide/stories/Typography/Text.stories.mdx (hunk 7 above)
Comment (codecaaron) [reply packages/styleguide/stories/Typography/Text.stories.mdx:125]: Yeah, not a ticket proper but there is a [note in the migration doc](https://www.notion.so/codecademy/Variance-Migration-a675bbc31b584fec9dfff72c0364b194#371aa6a1abee440b8cd3c2404419c72f).

The current enhancers read the old system props, I believe switching them over will be pretty painless (hopefully).

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.39-alpha.efc620.0<br/>@codecademy/gamut-labs@14.0.1-alpha.efc620.0<br/>@codecademy/gamut@32.0.1-alpha.efc620.0<br/>@codecademy/styleguide@36.0.1-alpha.efc620.0<br/></pre>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.39-alpha.b7c305.0<br/>@codecademy/gamut-labs@14.0.1-alpha.b7c305.0<br/>@codecademy/gamut@32.0.1-alpha.b7c305.0<br/>@codecademy/styleguide@36.0.1-alpha.b7c305.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60806871b6e8d12ac7967239--gamut-preview.netlify.app>https://60806871b6e8d12ac7967239--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60806871b6e8d12ac7967239>Deploy Logs</a></p>

File: packages/gamut/src/Typography/variants.ts (hunk 4 above)
Comment (codecaaron) [reply packages/gamut/src/Typography/variants.ts:2]: Concur.  Let me ask Yassine, should be good :)

Comment (codecaaron) [reply packages/gamut/src/Typography/variants.ts]: Went ahead and updated.

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.39-alpha.4539a6.0<br/>@codecademy/gamut-labs@14.0.1-alpha.4539a6.0<br/>@codecademy/gamut@32.0.1-alpha.4539a6.0<br/>@codecademy/styleguide@36.0.1-alpha.4539a6.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60807f815dedb900bd76b223--gamut-preview.netlify.app>https://60807f815dedb900bd76b223--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60807f815dedb900bd76b223>Deploy Logs</a></p>

File: packages/gamut-labs/src/GlobalFooter/__tests__/GlobalFooter-test.tsx (hunk 2 above)
Comment (JoshuaKGoldberg) [reply packages/gamut-labs/src/GlobalFooter/__tests__/GlobalFooter-test.tsx:11]: bamp @codecaaron - thoughts on renaming the test? (definitely a nit, not blocking)

**Review by JoshuaKGoldberg — APPROVED** [review]
Glorious!! 💯

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://60830e54baaec00091595303--gamut-preview.netlify.app>https://60830e54baaec00091595303--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/60830e54baaec00091595303>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.44-alpha.147708.0<br/>@codecademy/gamut-labs@14.2.4-alpha.147708.0<br/>@codecademy/gamut@32.1.4-alpha.147708.0<br/>@codecademy/styleguide@36.2.5-alpha.147708.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608314e9f57f200095cf53a5--gamut-preview.netlify.app>https://608314e9f57f200095cf53a5--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608314e9f57f200095cf53a5>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.44-alpha.6d7846.0<br/>@codecademy/gamut-labs@14.2.4-alpha.6d7846.0<br/>@codecademy/gamut@32.1.4-alpha.6d7846.0<br/>@codecademy/styleguide@36.2.5-alpha.6d7846.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6083164abaaec0009d5953bd--gamut-preview.netlify.app>https://6083164abaaec0009d5953bd--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6083164abaaec0009d5953bd>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.44-alpha.db274e.0<br/>@codecademy/gamut-labs@14.2.4-alpha.db274e.0<br/>@codecademy/gamut@32.1.4-alpha.db274e.0<br/>@codecademy/styleguide@36.2.5-alpha.db274e.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608317d7ff5c3200dc8b9baa--gamut-preview.netlify.app>https://608317d7ff5c3200dc8b9baa--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608317d7ff5c3200dc8b9baa>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.45-alpha.98b642.0<br/>@codecademy/gamut-labs@14.2.5-alpha.98b642.0<br/>@codecademy/gamut@32.1.4-alpha.98b642.0<br/>@codecademy/styleguide@36.2.6-alpha.98b642.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://608333904eaeae2bf73dcae8--gamut-preview.netlify.app>https://608333904eaeae2bf73dcae8--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/608333904eaeae2bf73dcae8>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.3.46-alpha.c78829.0<br/>@codecademy/gamut-labs@14.2.6-alpha.c78829.0<br/>@codecademy/gamut@32.1.5-alpha.c78829.0<br/>@codecademy/styleguide@36.2.7-alpha.c78829.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6086b9b24165ec89a656c8a6--gamut-preview.netlify.app>https://6086b9b24165ec89a656c8a6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6086b9b24165ec89a656c8a6>Deploy Logs</a></p>

**Pull request merged**